examples/Simple/TopSort.hs

```diff
@@ -0,0 +1,115 @@
+{-# OPTIONS_GHC -Wno-name-shadowing #-}
+{-# OPTIONS_GHC -Wno-unused-matches #-}
+{-# LANGUAGE LinearTypes #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LambdaCase #-}
+
+
+module Simple.TopSort where
+
+import qualified Prelude.Linear as Linear
+import Prelude.Linear ((&))
+import Data.Unrestricted.Linear
+import qualified Data.HashMap.Linear as HMap
+import Data.HashMap.Linear (HashMap)
+import Data.Bifunctor.Linear (second)
+import Data.Maybe.Linear (catMaybes)
+import qualified Data.Functor.Linear as Data
+
+
+------------------------------
+
+test1 :: [Int]
+test1 = topsort
+  [(1,[2,3]), (2, [4]), (3,[4]), (4,[])]
+-- [1,2,3,4]
+
+test2 :: [Int]
+test2 = topsort
+  [(5,[2,0]), (4,[0,1]), (0,[]), (2,[3]), (3,[1]), (1,[])]
+-- [5,2,3,4,0,1]
+
+test3 :: [Int]
+test3 = topsort
+  [ (1,[2]), (2,[4,5]), (3,[9,7]), (4,[7,8,10]), (5,[10]), (6,[10])
+  , (7,[]),(8,[]),(9,[]),(10,[])
+  ]
+-- [1,2,4,8,5,3,9,7,6,10]
```

Comment:
Maybe we should find a way to check the answers in tests?

Reply:
Doctests? _[Leo out]_

Reply:
I wasn't able to set up doctests with linear base last time -- too many issues came up and it became a time sink. I just set up HUnit since it's simple enough.

Reply:
Maybe later, then.